llvm/lib/Support/APInt.cpp

```diff
@@ -74,11 +66,14 @@ inline static unsigned getDigit(char cdigit, uint8_t radix) {
 
 
 void APInt::initSlowCase(uint64_t val, bool isSigned) {
-  U.pVal = getClearedMemory(getNumWords());
-  U.pVal[0] = val;
-  if (isSigned && int64_t(val) < 0)
-    for (unsigned i = 1; i < getNumWords(); ++i)
-      U.pVal[i] = WORDTYPE_MAX;
+  if (isSigned && int64_t(val) < 0) {
+    U.pVal = new WordType[getNumWords()];
+    U.pVal[0] = val;
+    memset(&U.pVal[1], 0xFF, sizeof(WordType) * (getNumWords() - 1));
+  } else {
+    U.pVal = new WordType[getNumWords()]();
+    U.pVal[0] = val;
```

Comment:
This doesn't look correct to me. Not that clearUnusedBits() only clears bits in the high order word, so you will leave everything between the high order word and the low order word uninitialized.

Reply:
Please note the use of the default-initialization constructor on line 74.

Reply:
Oh wow, so one of these lines is `new WordType[getNumWords()]` and the other is `new WordType[getNumWords()]()`. I completely missed that.

I agree that the code is correct, but it looks like an undesirable obfuscation to me. I'll leave it to other people to decide.

Reply:
Would you be more comfortable with `new WordType[getNumWords()](/* default-initialization */)`?